src/main/java/htsjdk/samtools/util/BlockGunzipper.java

```diff
@@ -43,10 +44,35 @@
  * @author alecw@broadinstitute.org
  */
 public class BlockGunzipper {
-    private final Inflater inflater = new Inflater(true); // GZIP mode
+    private static InflaterFactory defaultInflaterFactory = new InflaterFactory();
+    private final Inflater inflater;
     private final CRC32 crc32 = new CRC32();
     private boolean checkCrcs = false;
 
+    /**
+     * Create BlockGunziper using the provided inflaterFactory
+     * @param inflaterFactory
+     */
+    BlockGunzipper(InflaterFactory inflaterFactory) {
+        inflater = inflaterFactory.makeInflater(true); // GZIP mode
+    }
```

Comment:
There should be a no-arg constructor in this class as well that calls `defaultInflaterFactory.makeInflater(true)`